### PR TITLE
refactor: rename AvailabilityTitleView to UserStatusView

### PR DESCRIPTION
**Pull request by caldrian**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

A [future PR](https://github.com/wireapp/wire-ios/pull/916/files#diff-886b8c9e13918b59a414cc654e29d296eb70b742db69d36951743c67e4d372cf) the `AvailabilityTitleView` will be refactored in order to show relevant user specific infos. As a preparation this PR renames the `AvailabilityTitleView` to `UserStatusView`.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by theswiftway — APPROVED** [review]
LGTM 🚀

File: wire-ios-data-model/Source/Notifications/ObjectObserverTokens/UserChangeObserver.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+//
+
+import Foundation
+
+@objc(ZMUserChangeObserver)
+public protocol UserChangeObserver: NSObjectProtocol {
+    func userDidChange(_ changeInfo: UserChangeInfo)
+}
+
+public typealias ZMUserObserver = UserChangeObserver
```
Comment (markusfassbender) [inline wire-ios-data-model/Source/Notifications/ObjectObserverTokens/UserChangeObserver.swift:26]: Do we really need the `typealias` or can we just replace it with `UserChangeObserver`?

File: wire-ios/Wire-iOS/Sources/UserInterface/Availability/UserStatusView.swift [hunk 2]
```diff
@@ -21,53 +21,27 @@ import WireDataModel
...
+final class UserStatusView: TitleView, UserChangeObserver {
 
     // MARK: - Properties
 
     private let user: UserType
-    private var observerToken: Any?
-    private var options: Options
-
-    private let feedbackGenerator = UIImpactFeedbackGenerator(style: .medium)
+    private var observerToken: NSObjectProtocol?
```
Comment (markusfassbender) [inline wire-ios/Wire-iOS/Sources/UserInterface/Availability/UserStatusView.swift:30]: 👍

**Review by markusfassbender — APPROVED** [review]

File: wire-ios-data-model/Source/Notifications/ObjectObserverTokens/UserChangeObserver.swift (hunk 1 above)
Comment (caldrian) [reply wire-ios-data-model/Source/Notifications/ObjectObserverTokens/UserChangeObserver.swift:26]: I'll remove it in a follow-up pr.

**Pull request merged**